### PR TITLE
update appwrappers to v0.24.0

### PR DESCRIPTION
**Pull request by dgrove-oss**
Features
* Upgrade to Kueue 0.8.0
* Use Patch operations to update AppWrapper.Status
* Inject AppWrapper finalizer in WebHook

Bugs and Regressions
* Get dev mode (make run) working again
* account for all covered resources when a node is cordoned

Other Changes
* Logging improvements for development/debugging
* Upgrade to Kind v0.24.0
* Unit test improvement
* add failing job with retry to end-to-end tests
* improve e2e testing of failures
* need non-zero reset pause to reliably detect Resetting phase

Full Changelog: https://github.com/project-codeflare/appwrapper/compare/v0.23.0...v0.24.0

### CODE REVIEW

File: go.mod
Comment (dgrove-oss) [inline go.mod:41]: This is a bit of a hack... but the PR to upgrade the ODH fork to Kueue 0.8 isn't merged yet and the bits of the Kueue go API that AppWrapper depends on were not changed from 0.7 to 0.8. 

We can either hold this PR until https://github.com/opendatahub-io/kueue/pull/44 is merged and refer to the new tag here _or_ we can merge this now and then update the replace once the ODH PR merges.  I believe both work.

Comment (varshaprasad96) [reply go.mod:41]: I'm fine with merging this for now, and following up with an update when the sync PR in ODH gets merged.

**Review by varshaprasad96 — APPROVED** [review]
/lgtm
/approve

File: go.mod
Comment (dgrove-oss) [reply go.mod:41]: Sounds good.  

Now that you are in the owners file, I think you may need to /approve again to get the bot to add the label and proceed with the merge.

**Comment by dgrove-oss** [comment]
Needs @varshaprasad96 to /approve again (she wasn't in the owners file for the initial /approve).

**Review by varshaprasad96 — APPROVED** [review]
/approve

**Pull request merged**
